File: clients/Ruby-client/lib/solidb/cluster_client.rb
module SoliDB
  class ClusterClient
    def initialize(client)
      @client = client
    end

    def status
      @client.send_command("cluster_status", {})
    end

    def info
      @client.send_command("cluster_info", {})
    end

    def remove_node(node_id)
      @client.send_command("cluster_remove_node", node_id: node_id)
      nil
    end

    def rebalance
      @client.send_command("cluster_rebalance", {})
      nil
    end

    def cleanup
      @client.send_command("cluster_cleanup", {})
      nil
    end

    def reshard(num_shards)
      @client.send_command("cluster_reshard", num_shards: num_shards)
      nil
    end

    def get_nodes
      @client.send_command("cluster_get_nodes", {}) || []
    end

    def get_shards
      @client.send_command("cluster_get_shards", {}) || []
    end
  end
end

File: clients/PHP-client/src/ClusterClient.php
<?php

namespace SoliDB;

class ClusterClient
{
    private Client $client;

    public function __construct(Client $client)
    {
        $this->client = $client;
    }

    public function status(): array
    {
        $res = $this->client->sendCommand('cluster_status', []);
        return $res['data'] ?? [];
    }

    public function info(): array
    {
        $res = $this->client->sendCommand('cluster_info', []);
        return $res['data'] ?? [];
    }

    public function removeNode(string $nodeId): void
    {
        $this->client->sendCommand('cluster_remove_node', ['node_id' => $nodeId]);
    }

    public function rebalance(): void
    {
        $this->client->sendCommand('cluster_rebalance', []);
    }

    public function cleanup(): void
    {
        $this->client->sendCommand('cluster_cleanup', []);
    }

    public function reshard(int $numShards): void
    {
        $this->client->sendCommand('cluster_reshard', ['num_shards' => $numShards]);
    }

    public function getNodes(): array
    {
        $res = $this->client->sendCommand('cluster_get_nodes', []);
        return $res['data'] ?? [];
    }

    public function getShards(): array
    {
        $res = $this->client->sendCommand('cluster_get_shards', []);
        return $res['data'] ?? [];
    }
}
